test/originals/embedding_test.rb

```diff
@@ -1,35 +1,48 @@
 require_relative './../test_helper'
 
 class EmbeddingTest < MiniTest::Test
-  
+
   def test_can_build_embedded_model
     restroom = Restroom.new
     door = restroom.build_door
     assert door.is_a?(Door)
   end
-  
+
   def test_ensure_door_returns_existing_door
     restroom = Restroom.new
     restroom.build_door
     restroom.door.open_handle = %w{knob}
     restroom.ensure_door
     assert_equal %w{knob}, restroom.door.open_handle
   end
-  
+
+  def test_ensure_change_events_are_propagated_to_root_has_many_has_one
```

Comment:
opinionated nit: this method name's quite long and I am not sure if the has_many_has_one part is relevant as we are testing just parent changes. Anyways, if we can do with a shorter, good, if not, no worries

Reply:
agreed